src/router/users-router.ts

```diff
@@ -0,0 +1,125 @@
+    makeUserUpdateDTO,
+    shortenUser,
+} from "../utils/utils";
+import {
+    validateUserFields,
+} from "../utils/validate-fields";
+import {responseHandlerHTML, responseHandlerJSON, showNotFoundView} from "./handler";
+import {IUser} from "../types/user-types";
+
+export const usersRouter = [
+    {
+        method: HTTP_METHODS.GET,
+        pattern: parserPatterns.users,
+        handler: async (req: IncomingMessage, res: ServerResponse) => {
+           try {
+               const users = userService.getAllUsers();
+               if(users.length){
+                   return responseHandlerJSON(res, HttpStatusCode.OK, users.map(shortenUser));
+               } else {
+                   return responseHandlerJSON(res, HttpStatusCode.NO_CONTENT, []);
+               }
+
+           } catch (err){
+               console.error(err);
+              return responseHandlerHTML(res, HttpStatusCode.BAD_REQUEST, errors.somethingWentWrong)
+           }
+        }
+    },
+    {
+        method: HTTP_METHODS.GET,
+        pattern: parserPatterns.userSubRoute,
+        handler: async (req: IncomingMessage, res: ServerResponse, id: string) => {
+            try {
+                const user = userService.getUserById(id);
+                const links = [
+                    {
+                        rel: 'hobbies',
+                        href: `/users/${id}/hobbies`,
+                        method: 'GET'
+                    }
+                ];
+                return responseHandlerJSON(res, HttpStatusCode.OK, {user: shortenUser(user!), links});
+            } catch (err) {
+                console.error(err);
+                return showNotFoundView(res, errors.unableFindUser(id));
+            }
+        }
+    },
+    {
+        method: HTTP_METHODS.POST,
+        pattern: /^\/users$/,
+        handler: async (req: IncomingMessage, res: ServerResponse) => {
+            try {
+                const body = await getRequestBody<IUserResponse>(req);
+                const user = makeUserDTO(body);
+                const invalidFields = validateUserFields(user);
+                console.log('body', body)
+                console.log('invalidFields', invalidFields)
+                if (invalidFields.length) {
+                    return responseHandlerJSON(res, HttpStatusCode.UNPROCESSABLE_ENTITY, invalidFields);
+                } else {
+                    try {
+                        const newUser = userService.createUser(user);
+                        return responseHandlerJSON(res, HttpStatusCode.CREATED, newUser);
+                    } catch (err) {
+                        console.error(err);
+                       return responseHandlerHTML(res, HttpStatusCode.CONFLICT, errors.userExists(body.email))
+                    }
+                }
+            } catch (err) {
+                console.error(err);
+                return responseHandlerHTML(res, HttpStatusCode.UNPROCESSABLE_ENTITY, errors.parsingError)
```

Comment:
any error in the code will produce the UNPROCESSABLE_ENTITY, but there are could be error in the logic itself - in this case we have to throw 500, not 422, because it can confuse and it will be very hard to debug.